src/coreclr/jit/valuenum.h

```diff
@@ -1527,6 +1650,110 @@ FORCEINLINE T ValueNumStore::SafeGetConstantValue(Chunk* c, unsigned offset)
     }
 }
 
+#if defined(FEATURE_SIMD)
+template <>
+FORCEINLINE simd8_t ValueNumStore::SafeGetConstantValue<simd8_t>(Chunk* c, unsigned offset)
+{
+    if (c->m_typ == TYP_SIMD8)
+    {
+        return reinterpret_cast<VarTypConv<TYP_SIMD8>::Lang*>(c->m_defs)[offset];
+    }
+    else
+    {
+        assert(false);
+        return {};
+    }
+}
+
+template <>
+FORCEINLINE simd16_t ValueNumStore::SafeGetConstantValue<simd16_t>(Chunk* c, unsigned offset)
+{
+    if (c->m_typ == TYP_SIMD16)
+    {
+        return reinterpret_cast<VarTypConv<TYP_SIMD16>::Lang*>(c->m_defs)[offset];
+    }
+    else
+    {
+        assert(false);
+        return {};
+    }
+}
+
+template <>
+FORCEINLINE simd32_t ValueNumStore::SafeGetConstantValue<simd32_t>(Chunk* c, unsigned offset)
+{
+    if (c->m_typ == TYP_SIMD32)
+    {
+        return reinterpret_cast<VarTypConv<TYP_SIMD32>::Lang*>(c->m_defs)[offset];
+    }
+    else
+    {
+        assert(false);
+        return {};
+    }
+}
+
+template <>
+FORCEINLINE simd8_t ValueNumStore::ConstantValueInternal<simd8_t>(ValueNum vn DEBUGARG(bool coerce))
+{
+    Chunk* c = m_chunks.GetNoExpand(GetChunkNum(vn));
+    assert(c->m_attribs == CEA_Const);
+
+    unsigned offset = ChunkOffset(vn);
+
+    if (c->m_typ == TYP_SIMD8)
+    {
+        assert(!coerce);
+        return SafeGetConstantValue<simd8_t>(c, offset);
+    }
+    else
+    {
+        assert(false);
+        return {};
+    }
+}
+
+template <>
+FORCEINLINE simd16_t ValueNumStore::ConstantValueInternal<simd16_t>(ValueNum vn DEBUGARG(bool coerce))
+{
+    Chunk* c = m_chunks.GetNoExpand(GetChunkNum(vn));
+    assert(c->m_attribs == CEA_Const);
+
+    unsigned offset = ChunkOffset(vn);
+
+    if (c->m_typ == TYP_SIMD16)
+    {
+        assert(!coerce);
+        return SafeGetConstantValue<simd16_t>(c, offset);
+    }
+    else
+    {
+        assert(false);
+        return {};
+    }
+}
+
+template <>
+FORCEINLINE simd32_t ValueNumStore::ConstantValueInternal<simd32_t>(ValueNum vn DEBUGARG(bool coerce))
+{
+    Chunk* c = m_chunks.GetNoExpand(GetChunkNum(vn));
+    assert(c->m_attribs == CEA_Const);
+
+    unsigned offset = ChunkOffset(vn);
+
+    if (c->m_typ == TYP_SIMD32)
+    {
+        assert(!coerce);
+        return SafeGetConstantValue<simd32_t>(c, offset);
+    }
+    else
+    {
+        assert(false);
+        return {};
```

Comment:
It would be nice to define a `simd8/16/32_t::Zero`-like `static constexpr` field (or function), semantics of `{ }` can be unclear to people not intimately familiar with the C++ initialization rules.